bitcoind-tests/tests/setup/test_util.rs

```diff
@@ -86,8 +86,8 @@ fn setup_keys(
     let mut x_only_keypairs = vec![];
     let mut x_only_pks = vec![];
 
-    for i in 0..n {
-        let keypair = bitcoin::secp256k1::Keypair::from_secret_key(&secp_sign, &sks[i]);
+    for sk in sks.iter().take(n) {
```

Comment:
In b328736efdc0e07e8605e37c1b7526e6b6542113:

`iter` and `take` continue to be unnecessary

Reply:
Its like I didn't even read your review the first time :(

Reply:
Will fix, and lol "continue to be unnecessary" - love it.

Reply:
I don't know what was the original intention of the code but as it reads the `take(n)` is required because `n` is a function parameter so we can't just iterate over all of `sk`.

Reply:
In the iteration above we construct `sk` by looping `n` times and pushing stuff onto `sk`.

We could add a `debug_assert` that `sk.len()` equals `n` if it makes you nervous.

Reply:
My bad, I did not read the code properly. Used `for sk in sks {` as suggested.